### PR TITLE
Make the tests not leak artery files

### PR DESCRIPTION
**Pull request by johanandren**
Should help figure out #21484

### CODE REVIEW

**Comment by johanandren** [comment]
(based on top of the flight recorder output fix)

File: akka-remote/src/main/scala/akka/remote/artery/ArteryTransport.scala [hunk 1]
```diff
@@ -436,8 +433,14 @@ private[remote] class ArteryTransport(_system: ExtendedActorSystem, _provider: R
   private def startMediaDriver(): Unit = {
     if (settings.Advanced.EmbeddedMediaDriver) {
       val driverContext = new MediaDriver.Context
-      if (settings.Advanced.AeronDirectoryName.nonEmpty)
+      if (settings.Advanced.AeronDirectoryName.nonEmpty) {
         driverContext.aeronDirectoryName(settings.Advanced.AeronDirectoryName)
+      } else {
+        // create a random name but include the actor system name for easier debugging
+        val uniquePart = UUID.randomUUID().toString
+        val randomName = s"${CommonContext.AERON_DIR_PROP_DEFAULT}-${system.name}-$uniquePart"
```
Comment (patriknw) [inline akka-remote/src/main/scala/akka/remote/artery/ArteryTransport.scala]: I wonder why our files are located in `/run/shm`. In Aeron sources I only see `/dev/shm`

**Comment by patriknw** [comment]
LGTM

**Review by patriknw — APPROVED** [review]

File: akka-remote/src/main/scala/akka/remote/artery/ArteryTransport.scala (hunk 1 above)
Comment (johanandren) [inline akka-remote/src/main/scala/akka/remote/artery/ArteryTransport.scala]: `/dev/shm` is a link to `/run/shm/`on the jenkins server (ubuntu?), and `/var/run` points to `/run`

Comment (patriknw) [inline akka-remote/src/main/scala/akka/remote/artery/ArteryTransport.scala]: ah, thanks

File: akka-remote-tests/src/multi-jvm/scala/akka/remote/AttemptSysMsgRedeliverySpec.scala [hunk 2]
```diff
@@ -24,7 +24,7 @@ class AttemptSysMsgRedeliveryMultiJvmSpec(artery: Boolean) extends MultiNodeConf
   commonConfig(debugConfig(on = false).withFallback(
     ConfigFactory.parseString(s"""
       akka.remote.artery.enabled = $artery
-      """)).withFallback(MultiNodeRemotingSpec.arteryFlightRecordingConf))
+      """)).withFallback(RemotingMultiNodeSpec.arteryFlightRecordingConf))
```
Comment (johanandren) [inline akka-remote-tests/src/multi-jvm/scala/akka/remote/AttemptSysMsgRedeliverySpec.scala]: Figured out that the testkit will ignore classes that end with `MultiNodeSpec` so this fixes the actor system name breakage I caused by introducing an abstract base class last week.

Comment (patriknw) [inline akka-remote-tests/src/multi-jvm/scala/akka/remote/AttemptSysMsgRedeliverySpec.scala]: ok

File: akka-remote/src/main/scala/akka/remote/artery/ArteryTransport.scala [hunk 3]
```diff
@@ -427,17 +426,22 @@ private[remote] class ArteryTransport(_system: ExtendedActorSystem, _provider: R
   private lazy val shutdownHook = new Thread {
     override def run(): Unit = {
       if (!_shutdown) {
-        internalShutdown()
-
+        Await.result(internalShutdown(system.dispatcher), 20.seconds)
```
Comment (johanandren) [inline akka-remote/src/main/scala/akka/remote/artery/ArteryTransport.scala]: Actual leak fix

File: akka-remote/src/main/scala/akka/remote/artery/ArteryTransport.scala [hunk 4]
```diff
@@ -745,10 +748,10 @@ private[remote] class ArteryTransport(_system: ExtendedActorSystem, _provider: R
         flushingPromise.future
       }
     implicit val ec = remoteDispatcher
-    flushing.recover { case _ ⇒ Done }.flatMap(_ ⇒ internalShutdown())
+    flushing.recover { case _ ⇒ Done }.flatMap(_ ⇒ internalShutdown(system.dispatcher))
```
Comment (johanandren) [inline akka-remote/src/main/scala/akka/remote/artery/ArteryTransport.scala]: damn, I left this after thinking the ec was the problem, I'll remove it.

File: akka-remote/src/main/scala/akka/remote/artery/FlightRecorderReader.scala [hunk 5]
```diff
@@ -199,7 +199,7 @@ private[akka] final class FlightRecorderReader(fileChannel: FileChannel) {
   case class RichEntry(timeStamp: Instant, dirty: Boolean, code: Long, metadata: Array[Byte]) {
     override def toString: String = {
       val textualCode = FlightRecorderEvents.eventDictionary.getOrElse(code, "").take(34)
-      val metadataString = new String(metadata, "UTF-8")
+      val metadataString = new String(metadata, "US-ASCII")
```
Comment (johanandren) [inline akka-remote/src/main/scala/akka/remote/artery/FlightRecorderReader.scala]: realized it was written as ascii

**Comment by patriknw** [comment]
LGTM

File: akka-remote/src/main/scala/akka/remote/artery/FlightRecorderReader.scala (hunk 5 above)
Comment (drewhk) [inline akka-remote/src/main/scala/akka/remote/artery/FlightRecorderReader.scala]: Yes

**Comment by drewhk** [comment]
LGTM

**Pull request merged**